lib/generic-parser.js

```diff
@@ -0,0 +1,160 @@
+/*!
+ * body-parser
+ * Copyright(c) 2014 Jonathan Ong
+ * Copyright(c) 2014-2015 Douglas Christopher Wilson
+ * MIT Licensed
+ */
+
+'use strict'
+
+/**
+ * Module dependencies.
+ * @private
+ */
+
+var bytes = require('bytes')
+var contentType = require('content-type')
+var createError = require('http-errors')
+var debug = require('debug')('body-parser:generic')
+var isFinished = require('on-finished').isFinished
+var read = require('./read')
+var typeis = require('type-is')
+
+/**
+ * Module exports.
+ */
+
+module.exports = generic
+
+/**
+ * Use this to create a middleware that parses request bodies
+ *
+ * @param {object} [options]
+ * @return {function}
+ * @public
+ */
+
+function generic (parserOptions, parserOverrides) {
+  // Squash the options and the overrides down into one object
+  var opts = Object.create(parserOptions)
+  Object.assign(opts, parserOverrides)
+
+  var limit = typeof opts.limit !== 'number'
+    ? bytes.parse(opts.limit || '100kb')
+    : opts.limit
+  var charset = opts.charset
+  var inflate = opts.inflate !== false
+  var verify = opts.verify || false
+  var parse = opts.parse || defaultParse
```

Comment:
I'm not in favor of the `defaultParse` - it should be a mandatory option instead.